File: Codellama-34b-Instruct/Template_and_context_final/acc_set_default_async.cpp
#include "acc_testsuite.h"

#ifndef T1
//T1:acc set default async,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    // Set the default asynchronous activity queue to the initial value
    acc_set_default_async(acc_async_default);

    // Create a new asynchronous activity queue
    int async_arg = acc_async_create();

    // Set the default asynchronous activity queue to the new queue
    acc_set_default_async(async_arg);

    // Create a compute construct with an async clause
    #pragma acc parallel async
    {
        // Do some work
    }

    // Check that the work was done in the new queue
    if (acc_get_async_queue() != async_arg) {
        err = 1;
    }

    // Reset the default asynchronous activity queue to the initial value
    acc_set_default_async(acc_async_default);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Codellama-34b-Instruct/Template-and-context-final/acc_set_default_async.f90
#ifndef T1
!T1:acc set default async,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: async_arg
        INTEGER :: async_default
        
        ! Set the default asynchronous activity queue to a non-default value
        async_default = acc_async_default
        async_arg = acc_async_noval
        CALL acc_set_default_async(async_arg)
        
        ! Check that the default asynchronous activity queue has been set correctly
        IF (acc_default_async_var .NE. async_arg) THEN
          errors = errors + 1
        END IF
        
        ! Reset the default asynchronous activity queue to its initial value
        CALL acc_set_default_async(async_default)
        
        ! Check that the default asynchronous activity queue has been reset correctly
        IF (acc_default_async_var .NE. async_default) THEN
          errors = errors + 1
        END IF
        
        IF (errors .EQ. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif